src/main/java/com/spikes2212/command/drivetrains/commands/DriveArcadeWithPID.java

```diff
@@ -1,61 +1,91 @@
 package com.spikes2212.command.drivetrains.commands;
 
 import com.spikes2212.command.drivetrains.TankDrivetrain;
-import com.spikes2212.control.PIDLoop;
+import com.spikes2212.control.PIDSettings;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.controller.PIDController;
 import edu.wpi.first.wpilibj2.command.CommandBase;
 
 import java.util.function.Supplier;
 
-
+/**
+ * A command that moves a drivetrain with a set speed forward and with a PID loop to a certain angle.
+ */
 public class DriveArcadeWithPID extends CommandBase {
-
+    /**
+     * The drivetrain this command operates on.
+     */
     private final TankDrivetrain drivetrain;
-    private final PIDLoop pidLoop;
+
+    /**
+     * The PID Settings for the turning PID loop.
+     */
+    private PIDSettings pidSettings;
+
+    /**
+     * The PID Controller for the turning PID loop.
+     */
+    private PIDController pidController;
+
+    /**
+     * The angle of the drivetrain.
+     */
+    private Supplier<Double> source;
+
+    /**
+     * The last time the drivetrain's angle wasn't within the target range.
+     */
+    private double lastTimeNotOnTarget;
+
+    /**
+     * The angle the drivetrain should reach.
+     */
     private Supplier<Double> setpoint;
 
     /**
-     * @param drivetrain is the {@link TankDrivetrain} the command moves.
-     * @param pidLoop is the {@link PIDLoop} that calculates and sets the speed to the drivetrain.
+     * The speed at which to move the drivetrain forward.
      */
+    private Supplier<Double> moveValue;
 
-    public DriveArcadeWithPID(TankDrivetrain drivetrain, PIDLoop pidLoop, Supplier<Double> setpoint) {
+    public DriveArcadeWithPID(TankDrivetrain drivetrain, PIDSettings pidSettings, Supplier<Double> source,
```

Comment:
please add optional support for feed forward to this command as described above.